api/types/events/api.go

```diff
@@ -71,10 +71,24 @@ type AuditEvent interface {
 	SetClusterName(string)
 }
 
+type AuditContext struct {
+	context.Context
+}
+
+func NewAuditContext(ctx context.Context) AuditContext {
```

Comment:
I'm not sure how this is any better than the original. It still doesn't prevent you from using a request-scoped context - you can easily do NewAuditContext(r.Context()) and you have the same problem but more  non-standard types.

Reply:
I thought the goal was to avoid the audit context getting closed with the request, with this we avoid that problem while also keeping around the context for its data, which was the idea that i gathered from this comment https://github.com/gravitational/teleport-private/issues/873#issuecomment-1805839004

Changing the type means it has at least some indication of an error at compile time that it shouldnt just be the request local context.

Reply:
Assuming we are keeping the ctx parameter in audit methods so it can carry traces and the like, then the request context is the correct context to pass along (we'll disregard any deadlines it carries).